src/python/pants/subsystem/subsystem.py

```diff
@@ -105,7 +107,7 @@ def scoped_instance(cls: Type[_S], optionable: Optionable) -> _S:
     :param optionable: An optionable type or instance to scope this subsystem under.
     :returns: The scoped subsystem instance.
     """
-    if not isinstance(optionable, Optionable) and not issubclass(optionable, Optionable):  # type: ignore[misc]
+    if not isinstance(optionable, Optionable) and not issubclass(optionable, Optionable):  # type: ignore[misc] # MyPy thinks the RHS never evaluates, but it somehow does
```

Comment:
I think this is actually pointing out broken code or typing. The claim in the function signature typing is that `optionable` is an instance of `Optionalble`, so it's certainly not a Type which the issubclass checks.